version-ranges/src/lib.rs

```diff
@@ -826,6 +826,175 @@ impl<V: Ord + Clone> Ranges<V> {
     }
 }
 
+impl<V: Ord> FromIterator<(Bound<V>, Bound<V>)> for Ranges<V> {
+    /// Constructor from arbitrary, unsorted and potentially overlapping ranges.
+    ///
+    /// This is equivalent, but faster, to computing the [`Ranges::union`] of the
+    /// [`Ranges::from_range_bounds`] of each segment.
+    fn from_iter<T: IntoIterator<Item = (Bound<V>, Bound<V>)>>(iter: T) -> Self {
+        // We have three constraints we need to fulfil:
+        // 1. The segments are sorted, from lowest to highest (through `Ord`): By sorting.
+        // 2. Each segment contains at least one version (start < end): By `union`.
+        // 3. There is at least one version between two segments: By `union`.
```

Comment:
"By `union`" does not match the implementation. While I'm making nits about comments, let's  make this clear.

Reply:
That's a good catch, i forgot to update the comment after changing the implementation.